src/netaddress.cpp

```diff
@@ -328,28 +480,65 @@ enum Network CNetAddr::GetNetwork() const
     return m_net;
 }
 
+static std::string IPv6ToString(Span<const uint8_t> a) {
+    assert(a.size() == ADDR_IPV6_SIZE);
+    return strprintf("%x:%x:%x:%x:%x:%x:%x:%x",
+                     a[0] << 8 | a[1], a[2] << 8 | a[3],
+                     a[4] << 8 | a[5], a[6] << 8 | a[7],
+                     a[8] << 8 | a[9], a[10] << 8 | a[11],
+                     a[12] << 8 | a[13], a[14] << 8 | a[15]);
```

Comment:
1046e17c nit, this might be more readable with one element per line:
```diff
-                     a[0] << 8 | a[1], a[2] << 8 | a[3],
-                     a[4] << 8 | a[5], a[6] << 8 | a[7],
-                     a[8] << 8 | a[9], a[10] << 8 | a[11],
-                     a[12] << 8 | a[13], a[14] << 8 | a[15]);
+                      a[0] << 8 | a[1],
+                      a[2] << 8 | a[3],
+                      a[4] << 8 | a[5],
+                      a[6] << 8 | a[7],
+                      a[8] << 8 | a[9],
+                     a[10] << 8 | a[11],
+                     a[12] << 8 | a[13],
+                     a[14] << 8 | a[15]);
```

Reply:
Changed to `ReadBE16()`, must be more readable now.